parser-sdk/nodejs/parser-utils.js

```diff
@@ -0,0 +1,41 @@
+const fs = require("fs");
+const Ajv = require("ajv-draft-04");
+const ajv = new Ajv();
+const util = require("util");
+// eslint-disable-next-line security/detect-non-literal-fs-filename
+const readFile = util.promisify(fs.readFile);
+const { v4: uuid } = require("uuid");
+
+function addIdsAndDates(findings) {
+  return findings.map((finding) => {
+    return {
+      ...finding,
+      id: uuid(),
+      parsed_at: new Date().toISOString(),
+    };
+  });
+}
+
+// eslint-disable-next-line
+async function validate(jsonData) {
+  const jsonSchemaString = await readFile(
+    __dirname + "/findings-schema.json",
+    "utf8"
+  );
+  const jsonSchema = JSON.parse(jsonSchemaString);
+  const validater = ajv.compile(jsonSchema);
+  const valid = validater(jsonData);
+  if (!valid) {
+    throw new Error(JSON.stringify(validater.errors, null, 2));
```

Comment:
```suggestion
  const validator = ajv.compile(jsonSchema);
  const valid = validator(jsonData);
  if (!valid) {
    throw new Error(JSON.stringify(validator.errors, null, 2));
```

Reply:
If I want to commit this, do I have to manually add the Signed-Off tag?

Reply:
I'm not sure what happens in this case. I would just try to commit it without any manual changes and look what happens. If the Bot complains, you could always rebase that commit.